### ISSUE TITLE
Add highlighting for variables

### ISSUE THREAD

**Issue by LoadingBG** [rhysd/vim-crystal#106]
The problem I have with the highlighting is that variables blend with the code:
```cr
puts arr
```
is just one color - grey, which can be confusing:
```
puts cr # How it looks in neovim - same color
```
Also variables within anonymous functions are colored just when they're defined:
```cr
10.times { |x| puts x }
```
only the `x` between the pipes is colored while the second use of `x` is not.
Here's [a picture](https://imgur.com/a/ZNHfCUI) of a test program and the highlighting I have.

**Comment by jlcrochet** [comment]
Local variables aren't highlighted by default because some people don't like them to be; however, they are matched as `crystalLocalVariableOrMethod`, so you can try `:hi! link crystalLocalVariableOrMethod Identifier` or something similar based on your preferences.

**Comment by LoadingBG** [comment]
The command doesn't work for me.

**Comment by jlcrochet** [comment]
You're right, I forgot that `crystalLocalVariableOrMethod` is transparent.

**Comment by rhysd** [comment]
I think highlighting variables perfectly is not possible with highlighting system of Vim/Neovim. Since it requires Crystal parser. Until parsing the source into Crystal abstract syntax tree, it cannot be determined that the identifier is variable or not. Neovim's tree-sitter support may change the situation in the future, though.
